shardingsphere-jdbc/shardingsphere-jdbc-core/src/main/java/org/apache/shardingsphere/driver/jdbc/core/statement/ShardingSphereStatement.java

```diff
@@ -185,19 +184,22 @@ private static SQLFederationDeciderContext decide(final LogicSQL logicSQL, final
         return deciderEngine.decide(logicSQL, database);
     }
     
-    private TrafficContext getTrafficContext(final LogicSQL logicSQL) {
-        TrafficContext result = TrafficContextHolder.get().orElseGet(() -> createTrafficContext(logicSQL));
-        if (connection.isHoldTransaction()) {
-            TrafficContextHolder.set(result);
+    private Optional<String> getInstanceIdAndSave(final LogicSQL logicSQL) {
+        Optional<String> trafficInstanceId = connection.getSqlSession().getTrafficInstanceId();
+        if (!trafficInstanceId.isPresent()) {
+            trafficInstanceId = getTrafficInstanceId(logicSQL);
         }
-        return result;
+        if (connection.isHoldTransaction() && trafficInstanceId.isPresent()) {
+            connection.getSqlSession().setTrafficInstanceId(trafficInstanceId.get());
+        }
+        return trafficInstanceId;
```

Comment:
Please rename trafficInstanceId to result.

Reply:
done.Thanks